src/core/lib/surface/channel_ping.cc

```diff
@@ -57,7 +57,7 @@ void grpc_channel_ping(grpc_channel* channel, grpc_completion_queue* cq,
   pr->tag = tag;
   pr->cq = cq;
   GRPC_CLOSURE_INIT(&pr->closure, ping_done, pr, grpc_schedule_on_exec_ctx);
-  op->send_ping = &pr->closure;
+  op->send_ping.on_ack = &pr->closure;
```

Comment:
we don't set the other field?

Reply:
`grpc_channel_ping()` is a public interface that does not support `on_initiate`. Currently there is no need to add `on_initiate` here. As `op` is filled with 0s in `grpc_make_transport_op()`, we don't have to assign nullptr to `op->send_ping.on_initiate`.